core/src/main/java/org/infinispan/interceptors/BaseCustomAsyncInterceptor.java

```diff
@@ -21,7 +21,7 @@
  * @author Dan Berindei
  * @since 9.0
  */
-public class BaseCustomSequentialInterceptor extends DDSequentialInterceptor {
+public class BaseCustomAsyncInterceptor extends DDAsyncInterceptor {
    protected Cache<?, ?> cache;
    protected EmbeddedCacheManager embeddedCacheManager;
 
```

Comment:
the ```setup``` condition is wrong. it should check if cache and the cache manager is not null but it is checking the cache twice.